### PR TITLE
Follow-up for HiDream-O1

### PR DESCRIPTION
**Pull request by kohya-ss**
Follow-up for #947.

Work in progress. Targeting `dev`.

- [x] Fix bucket registration & gradient checkpointing (make it work)
- [x] Fold DINO loss into `compute_loss`, drop `apply_auxiliary_losses` seam
- [x] Revert prodigy detection to `endswith` (needs intent check w/ @sdbds)
- [x] Remaining review items (qinglong docs, fp8_scaled placement, spec docs, etc.)
---

This pull request updates the HiDream-O1 documentation to clarify and improve instructions for both LoRA and full finetuning workflows, and removes the now-redundant full finetune spec file. The changes focus on clarifying the usage of training tasks, control/reference datasets, LoRA conv-layer adaptation, memory options, and timestep sampling ratios. The documentation is now more explicit about required arguments and behaviors, especially for control (i2i) and text-to-image (t2i) tasks.

**Key documentation improvements and removals:**

**1. HiDream-O1 LoRA and full finetune documentation updates**  
* Clarified the use of `--task t2i` (text-to-image) and `--task i2i` (image-to-image) for both LoRA and full finetuning, including how task selection affects which model components are trained and which datasets are compatible.  
* Added explicit instructions and example commands for i2i LoRA training, including how to adapt conv layers with `--network_args conv_dim=4 conv_alpha=1`, and clarified that conv-layer LoRA is opt-in and must be set deliberately.  
* Updated memory-related notes, especially about block-swap requiring batch size 1 and its incompatibility with larger batches.  
* Improved explanations of LoRA and full finetune target layers, and clarified the relationship between dataset type, task argument, and which model parts are frozen or trained.  
* Provided more explicit guidance on cache rebuilding when switching dataset types or tasks.  
[[1]](diffhunk://#diff-b911033fa5483252e17a8871240475a48603481e353173507dcc95fc0b5995fcL96-R94) [[2]](diffhunk://#diff-b911033fa5483252e17a8871240475a48603481e353173507dcc95fc0b5995fcR105-R134) [[3]](diffhunk://#diff-b911033fa5483252e17a8871240475a48603481e353173507dcc95fc0b5995fcL148-R164) [[4]](diffhunk://#diff-b911033fa5483252e17a8871240475a48603481e353173507dcc95fc0b5995fcL165-L173)

**2. Timestep sampling ratio documentation**  
* Updated both English and Japanese advanced config docs to reflect new sampler ratios for `qinglong_flux` and `qinglong_qwen`, including a table summarizing the ratios and a note about the HiDream-O1 update.  
* Clarified that `qinglong_qwen` skips the `logsnr` bucket and uses a Qwen-Image-specific shift.  
[[1]](diffhunk://#diff-a85869058d90b7d68afbfbdc1a1313d1b3f4244cab3c3a362555873a68708234L627-R638) [[2]](diffhunk://#diff-a85869058d90b7d68afbfbdc1a1313d1b3f4244cab3c3a362555873a68708234L651-R671)

**3. Removal of outdated full finetune spec**  
* Deleted `docs/hidream_o1_full_finetune_spec.md`, as its content is now reflected in the main documentation and no longer needed.  
([docs/hidream_o1_full_finetune_spec.mdL1-L271](diffhunk://#diff-65080283d671574b99728519eff4161ef500ec784bedae2c702fca404c413cadL1-L271))

**Other minor documentation cleanups:**  
* Removed an obsolete reference to the i2i training spec from `docs/hidream_o1.md`.  
([docs/hidream_o1.mdL30-L31](diffhunk://#diff-b911033fa5483252e17a8871240475a48603481e353173507dcc95fc0b5995fcL30-L31))

These changes make the documentation more accurate and user-friendly, reducing the risk of configuration errors and making advanced options more discoverable.

### CODE REVIEW

**Comment by kohya-ss** [comment]
@sdbds 

I changed `generate_step_logs` in `trainer_base.py` back to `endswith` to avoid `prodigyplusschedulefree` entering the `d*lr` path (`prodigyplusschedulefree` matches both `in` and `endswith`.)

If there's a reason for the change, I'd appreciate it if you could tell me.

**Comment by sdbds** [comment]
> @sdbds
> 
> I changed `generate_step_logs` in `trainer_base.py` back to `endswith` to avoid `prodigyplusschedulefree` entering the `d*lr` path (`prodigyplusschedulefree` matches both `in` and `endswith`.)
> 
> If there's a reason for the change, I'd appreciate it if you could tell me.

To match the prodigy_adv optimizer.
Because 8-bit processing performed poorly on the zimage model in previous tests, I tested these accelerated versions of the advanced optimizer.
https://github.com/Koratahiu/Advanced_Optimizers

**Comment by kohya-ss** [comment]
> To match the prodigy_adv optimizer.
> Because 8-bit processing performed poorly on the zimage model in previous tests, I tested these accelerated versions of the advanced optimizer.
> https://github.com/Koratahiu/Advanced_Optimizers

Thank you for clarification. So it will be better to first check `prodigyplusschedulefree` first, and if not, check `prodigy`.

**Comment by kohya-ss** [comment]
@sdbds 

I noticed that specifying `fp8_base` for finetuning reduces parameter accuracy because the trainable parameters are first cast to FP8 and then cast back again. In addition, even with `fp8_base` specified, the only memory reduction effect seems to be from the vision model.

I'm considering removing these functions - the FP8 casting and the re-casting to BF16/FP32 functions for simplification. What do you think?

**Comment by sdbds** [comment]
> @sdbds
> 
> I noticed that specifying `fp8_base` for finetuning reduces parameter accuracy because the trainable parameters are first cast to FP8 and then cast back again. In addition, even with `fp8_base` specified, the only memory reduction effect seems to be from the vision model.
> 
> I'm considering removing these functions - the FP8 casting and the re-casting to BF16/FP32 functions for simplification. What do you think?

I think so. I experimentally added FP8 conversion before, but it doesn't seem to have much of an impact.

**Comment by kohya-ss** [comment]
> I think so. I experimentally added FP8 conversion before, but it doesn't seem to have much of an impact.

Thank you for your confirmation. It seems that fp8 mixed precision training could be effectively used with a different architecture in future.

File: src/musubi_tuner/training/parser_common.py [hunk 1]
```diff
@@ -373,7 +373,6 @@ def _add_lr_scheduler_args(parser: argparse.ArgumentParser) -> None:
 
 def _add_memory_args(parser: argparse.ArgumentParser) -> None:
     parser.add_argument("--fp8_base", action="store_true", help="use fp8 for base model / base modelにfp8を使う")
-    parser.set_defaults(fp8_scaled=False)
     # parser.add_argument("--full_fp16", action="store_true", help="fp16 training including gradients / 勾配も含めてfp16で学習する")
```
[the remark this answers is not in the document]
Comment (kohya-ss) [reply src/musubi_tuner/training/parser_common.py:380]: Fixed.

**Comment by kohya-ss** [comment]
I've made several updates, including adding `--task` argument and changing the default behavior of `conv_dim` (to match sd-scripts).
